service/tests/helpers/mod.rs

```diff
@@ -0,0 +1,106 @@
+use std::{
+    io::{BufRead, BufReader, Read},
+    process::{Command, Stdio},
+};
+
+use portpicker::pick_unused_port;
+
+pub struct PostgresInstance {
+    port: u16,
+    container: String,
+    password: String,
+}
+
+impl PostgresInstance {
+    pub fn new() -> Self {
+        let port = pick_unused_port().expect("could not find a free port for postgres");
+        let container = "postgres-shuttle-service-integration-test".to_string();
+        let password = "password".to_string();
+
+        let mut command = Command::new("docker");
+        command.args([
+            "run",
+            "--name",
+            &container,
+            "-e",
+            &format!("POSTGRES_PASSWORD={}", password),
+            "-p",
+            &format!("{}:5432", port),
+            "postgres:11", // Our Dockerfile image is based on buster which has postgres version 11
+        ]);
+
+        Self::wait_for_up(&mut command);
+
+        Self {
+            port,
+            container,
+            password,
+        }
+    }
+
+    pub fn get_uri(&self) -> String {
+        format!(
+            "postgres://postgres:{}@localhost:{}",
+            self.password, self.port
+        )
+    }
+
+    fn wait_for_up(command: &mut Command) {
+        let mut child = command
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()
+            .expect("failed to spawn a postgres container");
+        let mut stdout = child
+            .stdout
+            .take()
+            .expect("failed to get stdout of container");
+
+        let mut buf = [0; 2 << 12];
+        let mut current_pos = 0;
+
+        'wait: while let Ok(len) = stdout.read(&mut buf[current_pos..]) {
```

Comment:
Is this needed? It seems to only bubble the spawned process's stdout so this could be done with `Stdio::inherit()` (which I believe is the default with `spawn`?) with potentially breaking when we see a particular line of log.

Maybe better to `while not timed out { docker exec container pg_isready; check exit code; sleep(100ms); }`?